### ISSUE TITLE
Obscure error when datalad update -r

### ISSUE THREAD

**Issue by oesteban** [templateflow/templateflow#5]
```
$ datalad update -r .
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/datalad/templateflow> 
update(ok): /home/oesteban/datalad/templateflow (dataset)
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/datalad/templateflow/tpl-MNI152Lin> 
update(ok): /home/oesteban/datalad/templateflow/tpl-MNI152Lin (dataset)
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/datalad/templateflow/tpl-MNI152NLin2009cAsym> 
update(ok): /home/oesteban/datalad/templateflow/tpl-MNI152NLin2009cAsym (dataset)
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/datalad/templateflow/tpl-NKI> 
update(ok): /home/oesteban/datalad/templateflow/tpl-NKI (dataset)
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/datalad/templateflow/tpl-OASIS30ANTs> 
update(ok): /home/oesteban/datalad/templateflow/tpl-OASIS30ANTs (dataset)
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/datalad/templateflow/tpl-PNC> 
update(ok): /home/oesteban/datalad/templateflow/tpl-PNC (dataset)
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/datalad/templateflow/tpl-fMRIPrep> 
update(ok): /home/oesteban/datalad/templateflow/tpl-fMRIPrep (dataset)
[ERROR  ] AttributeError() [dataset.py:_flyweight_id_from_args:123] (AttributeError) 
```
cc @yarikoptic

**Comment by yarikoptic** [comment]
Is it still reproducing itself? 
What version of datalad. Be better to pay such against datalad right away - we would give some handy instructions such as to provide output of datalad wtf ;-)

**Comment by yarikoptic** [comment]
FWIW - filed https://github.com/datalad/datalad/issues/3097  would be interesting to discover what leads it to fail - so do not ruin this precious bug case ;)  may be even just `tar -czvf ~/precious.tgz /home/oesteban/datalad/templateflow` before doing anything else ;)

**Comment by oesteban** [comment]
Sorry, I didn't attribute datalad the issue. I think there's something off about the super-dataset or the subdatasets (or maybe my own git/git-annex settings).

This how I replicate it:

```
oesteban@dendrite:~$ datalad install https://github.com/templateflow/templateflow.git
[INFO   ] Cloning https://github.com/templateflow/templateflow.git [1 other candidates] into '/home/oesteban/templateflow'
[INFO   ]   Remote origin not usable by git-annex; setting annex-ignore
install(ok): /home/oesteban/templateflow (dataset)
oesteban@dendrite:~$ cd templateflow/
oesteban@dendrite:~/templateflow$ datalad update -r .
[INFO   ] Fetching updates for <Dataset path=/home/oesteban/templateflow>
update(ok): /home/oesteban/templateflow (dataset)
[ERROR  ] AttributeError() [dataset.py:_flyweight_id_from_args:123] (AttributeError)
```

It also happens with my old, working clone of templateflow as indicated above.

**Comment by yarikoptic** [comment]
"workaround" - don't specify `.` path.  The intention was to point to subdatasets, and specifying current dataset apparently caused this

**Comment by soichih** [comment]
I get this error even if I am not using `.`

```
/mnt/datalad/datasets.datalad.org$ datalad update -r adhd200
...
update(ok): /mnt/datalad/datasets.datalad.org/adhd200/RawDataBIDS/Pittsburgh (dataset)
[INFO   ] Fetching updates for <Dataset path=/mnt/datalad/datasets.datalad.org/adhd200/RawDataBIDS/WashU> 
update(ok): /mnt/datalad/datasets.datalad.org/adhd200/RawDataBIDS/WashU (dataset)
[INFO   ] Fetching updates for <Dataset path=/mnt/datalad/datasets.datalad.org/adhd200/Resources> 
update(ok): /mnt/datalad/datasets.datalad.org/adhd200/Resources (dataset)
[INFO   ] Fetching updates for <Dataset path=/mnt/datalad/datasets.datalad.org/adhd200/surfaces> 
update(ok): /mnt/datalad/datasets.datalad.org/adhd200/surfaces (dataset)
[ERROR  ] AttributeError() [dataset.py:_flyweight_id_from_args:123] (AttributeError) 
```

**Comment by yarikoptic** [comment]
ok, a fix for 0.11.x is in https://github.com/datalad/datalad/pull/3996 , after merge into 0.11.x, will propagate into master.  Please let's continue on that PR or https://github.com/datalad/datalad/issues/3097

**Comment by oesteban** [comment]
This seems to have been fixed. Closing.
